### DISCUSSION TITLE
Native Support for Google Cloud Spanner

### DISCUSSION THREAD

**Discussion by NOT-HAL9000** [openfga/community#179] [Ideas]
I'm wondering if you have considered adding Google Cloud Spanner support for OpenFGA? 

It has global strong transactional consistency, extremely performant, 99.999 SLA, and it has zero downtime schema changes (everything you want for OpenFGA). It is also more cost effective to run spanner than it is to run Cloud SQL (Much cheaper to run ever since Spanner [changed their pricing model](youtube.com/watch?v=m3mbOgjqQ7k) - $65 dollars a month for about 1k reads / second and 200 writes / second with no downtime upgrades for more).

We are thinking of using OpenFGA with cloud spanner through the [PGAdapter (postgres adapter)](https://cloud.google.com/spanner/docs/pgadapter-start), which is possible because it supports [pgx connections](https://cloud.google.com/spanner/docs/pg-pgx-connect), however that is extra overhead and we would love it if it was possible to use spanner with openfga without any sidecar proxy adapter infra to manage.  

Spanner + OpenFGA work really well together

![Screenshot 2023-06-13 at 5 28 31 pm](https://github.com/openfga/openfga/assets/70297791/321a828b-a722-4806-b951-6db776116c06)

**Comment by aaguiarz-uy** [comment]
Hi @NOT-HAL9000,

We have discussed it, but we don't have any concrete plans yet. 

An additional alternative to the ones you mentioned above could be for you to write a Cloud Spanner OpenFGA adapter. It should not be hard to do, however, we can't guarantee that we'll include that as part of the OpenFGA binary.

**Comment by NOT-HAL9000** [comment]
Thanks @aaguiarz-uy. Unfortunately we don't have much go experience so we will just have to rely on the PGAdapter.

**Comment by dank303** [comment]
I'm in the same boat (wanting to use openfga with spanner database).    I'm also attempting to use pgadapter, but have run across several issues with the openfga and goose migrations that have constructs not supported by that flow.  Even that level of support (not using grpc) would be helpful.
